### PR TITLE
Ten hardest steps in courses

### PR DESCRIPTION
**Pull request by vessemer**
Featch grades for each step in each course, then find ten hardest steps,
plot diagram (amount of passed, number of step in course).

### CODE REVIEW

File: examples/ten_hardest_steps_in_courses.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+for scr in scores:
+    plt.plot([v[0] for v in scr])
+
+final = []
+for course in scores: 
+    final += course
+
+hardest = sorted(final)[0:10]
+print (hardest)
```
Comment (mehanig) [inline examples/ten_hardest_steps_in_courses.py]: why having space between print and ( ? , usually all function syntax in python looks like `func(...)`

File: examples/ten_hardest_steps_in_courses.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            it = row['results']
+            scores = [(score[0] + 1 if it[name]['is_passed'] else score[0], it[name]['step_id']) for name, score in zip(names, scores)]
+        page += 1
+    return scores
+
+scores = []
+for course in courses:
+    scores += [scores_per_course(course)]
+    
+%matplotlib inline
```
Comment (mehanig) [inline examples/ten_hardest_steps_in_courses.py]: ? hm?
```File "ten_hardest_steps_in_courses.py", line 48
    %matplotlib inline
    ^
SyntaxError: invalid syntax```

**Comment by mehanig** [comment]
Doesn't work

**Comment by vessemer** [comment]
This work on ipython notebook, I just forget to remove %matplotlib inline
notebook comand,
the space in print (hardest)
because I used to write in python2.x print "some think"
Over all I very late learned about this internship, and may commit, you may
see, only at 4:30 sorry for that.

File: examples/ten_hardest_steps_in_courses.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return scores
+
+scores = []
+for course in courses:
+    scores += [scores_per_course(course)]
+    
+import matplotlib.pyplot as plt
+
+for scr in scores:
+    plt.plot([v[0] for v in scr])
```
Comment (mehanig) [inline examples/ten_hardest_steps_in_courses.py]: Why it's needed? 
The only response I see is print statement

Comment (vessemer) [inline examples/ten_hardest_steps_in_courses.py]: Well, again the problem is in my configuration, this changes will fix the mistake

**Comment by mehanig** [comment]
```Traceback (most recent call last):
  File "ten_hardest_steps_in_courses.py", line 50, in <module>
    pylab.plot([v[0] for v in scr])
AttributeError: 'module' object has no attribute 'plot'```

**Pull request closed without merging**